docs/Tech_Grants_2021.md

```diff
@@ -0,0 +1,128 @@
+
+![](figures/training_experiments_all.png)
+
+The average metrics of all models reveal that the best experiment is `remove stop words + clean strings + 42 training data` (experiment 5). This gives highest test recall and precision, and good EPMC evaluation metric.
+
+![](figures/training_experiments_average.png)
+
+It appears that the exclusion of the grant type data made the models perform worse.
+
+|Metric|Experiment 0|Experiment 4|Experiment 5|
+|---|---|---|---|
+|Test precision (average of all models) | 0.791|0.822|0.825|
+|Test recall (average of all models) | 0.829|0.833|0.848|
+|Test precision (42 data) (average of all models) | 0.797|0.822|-|
+|Test recall (42 data) (average of all models) | 0.867|0.844|-|
+|EPMC accuracy (average of all models) | 0.676|0.706|0.699|
+|RF accuracy (average of all models) | 0.514|0.524|0.510|
+
+Note: When using the 42 data in the training the evaluation using 42 data is the same as the test metrics. Thus the 42 data evaluation is only really interesting to see how well the 360 giving data translates to 42 data.
+
+
+## `grant_tagger.py` additions
+
+Thus `grant_tagger.py` was adapted to:
+- Remove stop words for count and tfidf vectorizers.
+- Improved string cleaning of the training data.
+- The random seed is set to 1. Earlier we tried to optimise the value picked for this, but then this would overfit to the test data - so this time it wasn't picked with any thought for optimisation.
+
+A new model training config was made for training models (`configs/train_model/2021.03.31.ini`) which also takes the 42 grant data as an input to get the grant texts from.
+
+## Performance
+
+I ran:
+```
+python nutrition_labels/grant_tagger.py --config_path configs/train_model/2021.03.31.ini
+```
+
+I evaluated how well each model extended to make predictions of tech grants on the RF and EPMC datasets by running:
+```
+nutrition_labels/grant_tagger_evaluation.py --model_config configs/train_model/2021.03.31.ini --epmc_file_dir data/processed/training_data/210329epmc/training_data.csv --rf_file_dir data/processed/training_data/210329rf/training_data.csv
+``` 
+This script also outputs the test metrics for each model in one csv which gives:
+
+| Date   | Vectorizer | Classifier  | f1    | precision_score | recall_score | EPMC accuracy | RF accuracy |
+|--------|------------|-------------|-------|-----------------|--------------|---------------|-------------|
+| 210331 | count      | naive_bayes | 0.837 | 0.74            | 0.962        | 0.791         | 0.557       |
+| 210331 | count      | SVM         | 0.813 | 0.871           | 0.762        | 0.588         | 0.4         |
+| 210331 | count      | log_reg     | 0.83  | 0.835           | 0.825        | 0.595         | 0.4         |
+| 210331 | tfidf      | naive_bayes | 0.819 | 0.713           | 0.962        | 0.824         | 0.657       |
```

Comment:
this seems to perform nice across, maybe mention it. it would be nice to highlight best performing scores per column in bold